### ISSUE TITLE
irrigation controller

### ISSUE THREAD

**Issue by hros** [rospogrigio/localtuya#340]
Hi
I have a Tuya irrigation controller, which is not in the list of the supported device types.
Is there a tutorial/demo as to how to add support for new device types?

**Comment by ultratoto14** [comment]
Check the [wiki](https://github.com/rospogrigio/localtuya/wiki/Qualifying-a-device) 👍

**Comment by hros** [comment]
thanks!!!
took some tweaking (mainly disconnecting other running services such as Google Home, and the Smart Life app) to get the DPs

I then chose for each irrigation zone a "switch" type, and lo and behold it works from Home Assistant!

I have another TUYA device (a water heater switch/timer) which is connected to Home Assistant using the Tuya integration.
I'm thinking of removing that integration and switching to LocalTuya

Just to make sure, if I want to make the connection local (no connection to Tuya cloud services), I cannot disconnect them from the tuya cloud since this integration is using the same protocol and keys.
So I have to somehow block the internet connection (as well as DNS connection) to these devices.
Thus, the smart-life app would not be able to access them, and they will be accessible only by home-assistant.
If I ever want to make a modification using the smart-life app, I would have to unblock the internet/dns connection of the devices.
Is that correct?

If so, how would you go about creating these network rules? I do not have a multi-VLAN network. My network is based on TP-Links's Deco Mesh.

**Comment by ultratoto14** [comment]
Hi @hros, it could be great to post your DPS dump and perhaps a config example to help other users.

You can use the tuya device locally without blocking the cloud access, it can be available on demande in case you need to change something from the app.
It seems that the blocking may break the normal behavior of the device, search the issues and PRs for more information.

About the network rules I cannot give any help. I'm also using Deco Mesh but only in AP mode.

**Comment by hros** [comment]
had a bit of a hiccup with my homeassistant setup, and had to reconfigure
will post the relevant data once things are back to normal

meanwhile could you please explain the difference between this integration (localtuya) and the tuya integration. specifically for devices that are recognized by the tuya integration.
it was my understanding that tuya requires that the devices connect with the tuya cloud, whereas localtuya can work even if the devices are limited to local communication over the home network

**Comment by ultratoto14** [comment]
The difference is mainly on speed, the official tuya integration is working with the cloud, where this one use a local connection. But devices are not limited to local connection by default, it's just if you want to block internet access from the device.

**Comment by hros** [comment]
regarding DPs for the irrigation controller, it turned out to be really simple '101'..'104' true/false switches which correspondingly open and close the water for the four available irrigation zones (I guess irrigation controllers with different number of zones follow the same logic).

I am now facing a new "challenge", namely configuring thermostats.
My thermostats' app is WarmMe which turns out to be a proprietary front end to the ubiquitous tuya protocol (this [post](https://community.home-assistant.io/t/warmme-intergration-tuya/239238/4) describes how to register the device with tuya / smart-life apps).
I would like to add this device to home assistant using the Local-Tuya integration.
I already have the device id, and device key, and I am clueless regarding the meaning of the DPs:
```
AVAILABLE DPS ARE [{
'101': True, 
'102': 210, 
'103': 'smart', 
'104': 0, 
'105': 210, 
'106': 0, 
'107': False, 
'108': False, 
'109': 0, 
'110': 3, 
'111': '0', 
'112': 0, 
'113': 0, 
'114': '1', 
'116': 0, 
'117': 157, 
'118': 'warming', 
'121': 300}]
```
I think that the "210" values represent temperature in tenth of a Celsius, that is 21.0°C

I can try and modify settings to see how the DPs change
Do you happen to have the settings of a similar device that will serve as a starting point in my reverse engineering attempts?

**Comment by hros** [comment]
btw, is it possible to add entities incrementally? that is after I add one or several entities to a device, add other entities?
It would be easier to work that way when debuggin/reverse-engineering the entities

**Comment by ultratoto14** [comment]
Some climate device are already supported. You may open another issue for that device and keep this one with irrigation controller.
Check the wiki, there's a procedure describing how to qualify a device, meaning like you said, take one feature, set different values and identify which DP is linked to which feature.
